tests/cbmc/proofs/Makefile.common

```diff
@@ -0,0 +1,285 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################
+# This Makefile is the base makefile for CBMC proofs in this repository
+# Consumers of this makefile are expected to set the relevant variables
+# e.g. ENTRY, CBMCFLAGS, UNWINDSET etc
+# and then include this makefile.  To see how to use this makefile, take a look
+# at any of the existing proofs in this repository.
+#
+# This makefile is designed so that all files created during a proof are
+# placed in the proof directory, or subdirectories of it.
+# In particular, all gotos are built into the "gotos" folder, and logs go into
+# the "logs" folder. This means that the main source directories are untouched by
+# doing a build, and should make it safe to simultaniously build multiple proofs,
+# including proofs with different configurations and -D defines.
+#
+# Dependency handling in this file may not be perfect.  It is recommended that
+# you "make clean" or "make veryclean" before rerunning a proof.
+
+SHELL=/bin/bash
+
+default: report
+
+# if Makefile.local exists, use it. This provides a way to override the defaults
+sinclude Makefile.local
```

Comment:
What is "sinclude"?

Reply:
It's described in the comment above. (`sinclude` won't error out if the file doesn't exist).

I was also confused when I first saw this. I suggest changing the comment to
```sh
# Use `sinclude` to include Makefile.local if it exists. This provides a way to override the defaults
```
Also, the readme implies that windows users might be running these proofs. But the majority of this makefile can only be parsed by GNU Make, or assumes Bash or Unix (shell redirection, tee and friends, etc). Is that okay? (I realise this might become a non-issue when we start using cmake...)

Reply:
Removed most of the references to windows. Changed to the proposed comment for sinclude